description/src/description.rs

```diff
@@ -0,0 +1,234 @@
+use anyhow::anyhow;
+use scale_info::{
+    form::PortableForm, Field, PortableRegistry, Type, TypeDef, TypeDefArray, TypeDefBitSequence,
+    TypeDefCompact, TypeDefPrimitive, TypeDefSequence, TypeDefTuple, TypeDefVariant, Variant,
+};
+
+use crate::transformer::Transformer;
+
+use super::formatting::format_type_description;
+
+pub fn type_description(type_id: u32, type_registry: &PortableRegistry) -> anyhow::Result<String> {
+    fn return_type_name_on_recurse(
+        _type_id: u32,
+        ty: &Type<PortableForm>,
+        _transformer: &Transformer<String>,
+    ) -> anyhow::Result<String> {
+        if let Some(type_name) = ty.path.ident() {
+            return Ok(type_name);
+        }
+        Err(anyhow!("Recursive type that did not get handled properly"))
+    }
+
+    let transformer = Transformer::new(
+        ty_description,
+        return_type_name_on_recurse,
+        (),
+        type_registry,
+    );
+    let description = transformer.resolve(type_id)?;
+    let formatted = format_type_description(&description);
+    Ok(formatted)
+}
+
+fn ty_description(
+    _type_id: u32,
+    ty: &Type<PortableForm>,
+    transformer: &Transformer<String>,
+) -> anyhow::Result<String> {
+    let ident = ty.path.ident().unwrap_or_default();
+    let prefix = type_def_prefix(&ty.type_def);
+    let type_def_description = type_def_type_description(&ty.type_def, transformer)?;
+    Ok(format!("{prefix}{ident}{type_def_description}"))
+}
+
+/// todo: clean this up
+fn type_def_prefix(type_def: &TypeDef<PortableForm>) -> &str {
+    match type_def {
+        TypeDef::Variant(_) => "enum ",
+        _ => "",
+    }
+}
+
+fn type_def_type_description(
+    type_def: &TypeDef<PortableForm>,
+    transformer: &Transformer<String>,
+) -> anyhow::Result<String> {
+    match type_def {
+        TypeDef::Composite(composite) => fields_type_description(&composite.fields, transformer),
+
+        TypeDef::Variant(variant) => variant_type_def_type_description(variant, transformer),
+        TypeDef::Sequence(sequence) => sequence_type_description(sequence, transformer),
+        TypeDef::Array(array) => array_type_description(array, transformer),
+        TypeDef::Tuple(tuple) => tuple_type_description(tuple, transformer),
+        TypeDef::Primitive(primitive) => primitive_type_description(primitive),
+        TypeDef::Compact(compact) => compact_type_description(compact, transformer),
+        TypeDef::BitSequence(bit_sequence) => {
+            bit_sequence_type_description(bit_sequence, transformer)
+        }
+    }
+}
+
+fn tuple_type_description(
+    tuple: &TypeDefTuple<PortableForm>,
+    transformer: &Transformer<String>,
+) -> anyhow::Result<String> {
+    let mut output = "(".to_string();
+    let mut iter = tuple.fields.iter().peekable();
+    while let Some(ty) = iter.next() {
```

Comment:
dumb question,

The code itself doesn't check that the tuple has at least two fields such `(a, b)` be a valid tuple in rust. Is that possible to end up with `(a)` here? 

AFAIU, `(a)` will just be type `a` wrapped in "an unused_parens"